src/api/schema/metrics/bytes_processed.rs

```diff
@@ -26,8 +26,38 @@ impl ProcessedBytesTotal {
     }
 }
 
-impl From<Metric> for ProcessedBytesTotal {
+impl From<Metric> for BytesProcessedTotal {
     fn from(m: Metric) -> Self {
         Self(m)
     }
 }
+
+pub struct ComponentBytesProcessedTotal {
+    name: String,
+    metric: Metric,
+}
+
+impl ComponentBytesProcessedTotal {
+    /// Returns a new `ComponentBytesProcessedTotal` struct, which is a GraphQL type. The
+    /// component name is hoisted for clear field resolution in the resulting payload
+    pub fn new(metric: Metric) -> Self {
+        let name = metric.tag_value("component_name").expect(
+            "Returned a metric without a `component_name`, which shouldn't happen. Please report.",
+        );
+
+        Self { name, metric }
+    }
+}
+
+#[Object]
+impl ComponentBytesProcessedTotal {
+    /// Component name
+    async fn name(&self) -> String {
```

Comment:
A borrow should be returned to avoid cloning.

```rust
async fn name(&self) -> &String { ... }
```

Reply:
Ah, cool. I wasn't sure if lifetimes were affected in Object wrappers, so I defaulted to cloning everything. This is great. I've done a sweep for all `.clone()`s, and updated to `&str` for string types 👍